nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -189,7 +189,92 @@ in
       };
 
       buildMachines = mkOption {
-        type = types.listOf types.attrs;
+        type = types.loaOf (submodule ({ config }: {
+          options = {
+            sshUser = mkOption {
+              type = types.nullOr types.string;
+              default = null;
+              description = ''
+                The user as who to SSH to the build machine.
+              '';
+            };
+            hostName = mkOption {
+              type = types.string;
+              description = ''
+                The hostname of the build machine.
```

Comment:
Maybe note this hostname has to resolve.

Reply:
what good can be hostname that doesn't resolve ?